File: blinkeye-framewk-aprobaciones-dominio/src/main/java/net/royal/spring/workflow/dominio/dto/DtoWfPlanificacion.java
package net.royal.spring.workflow.dominio.dto;

import java.util.ArrayList;
import java.util.List;

import net.royal.spring.framework.modelo.generico.DominioTransaccion;

public class DtoWfPlanificacion extends DominioTransaccion {
	private Integer transaccion;
	private List<DtoWfPlanificacionEtapa> lstEtapas;
	private DtoWfPlanificacionConfiguracion config;

	public DtoWfPlanificacion() {
		this.lstEtapas = new ArrayList<DtoWfPlanificacionEtapa>();
		this.config = new DtoWfPlanificacionConfiguracion();
	}

	public List<DtoWfPlanificacionEtapa> getLstEtapas() {
		return lstEtapas;
	}

	public void setLstEtapas(List<DtoWfPlanificacionEtapa> lstEtapas) {
		this.lstEtapas = lstEtapas;
	}

	public DtoWfPlanificacionConfiguracion getConfig() {
		return config;
	}

	public void setConfig(DtoWfPlanificacionConfiguracion config) {
		this.config = config;
	}

	public Integer getTransaccion() {
		return transaccion;
	}

	public void setTransaccion(Integer transaccion) {
		this.transaccion = transaccion;
	}

}

File: web-deteccion/src/app/royal/modulos/framework-workflow/dominio/dto/DtoWfPlanificacion.ts
import { DominioTransaccion } from "@framework/modelo/generico/DominioTransaccion";

export class DtoWfPlanificacion extends DominioTransaccion {
    transaccion: number;
    lstEtapas: DtoWfPlanificacionEtapa[] = [];
    config: DtoWfPlanificacionConfiguracion;
    constructor() {
        super();
        this.lstEtapas = [];
        this.config = new DtoWfPlanificacionConfiguracion();
    }
}
export class DtoWfPlanificacionConfiguracion {
    puedeVer: string;
    puedeEditar: string;
    puedeActualizar: string;
}
export class DtoWfPlanificacionEtapa {
    nivel: number;
    nombre: string;
    inicio: Date;
    fin: Date;
    planificacionid: number;
}